repo.c

```diff
@@ -668,11 +690,15 @@ filelist_loadcb (Pool     *pool,
       return 1;
     }
 
+  /* Cleanup old libsolv cache files (if any) */
+  remove_files_by_ext (cachedir, ".solvx");
```

Comment:
this is really bad idea. that means, if you use even same repo with same metadata, you always remove stuff which is unrelated. I think we should not do this…

Reply:
If I can load previous libsolv cache, I don't execute this code (see previous `if` with `return 1`).
I just remove the solv files in case I couldn't load previous cache for the same metadata, in which case the metadata has changed. If I don't do that, the libsolv cache will only grow because we'll create a new file every time the metadata changes for that repo.

Reply:
Unless it's not a problem that `.solv[x]` files are never removed.